### PR TITLE
Adds a TDP `SyncKeys` event

### PR DESCRIPTION
**Pull request by ibeckermayer**
This is used to sync the caps lock key between the local and remote machines, as well as reset the server key state to all keys up.

Corresponds to the RDP [2.2.8.1.1.3.1.1.5 Synchronize Event (TS_SYNC_EVENT)](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/6c5d0ef9-4653-4d69-9ba9-09ba3acd660f). ~Currently we're ignore num-lock, scroll-lock, and kana-lock, for expediency's sake.~

Changelog: Added support for synchronizing the values of the following toggle keys: Scroll Lock, Num Lock, Caps Lock, Kana Lock

### CODE REVIEW

File: lib/srv/desktop/rdp/rdpclient/Cargo.toml [hunk 1]
```diff
@@ -20,7 +20,7 @@ num-traits = "0.2.16"
 rand = { version = "0.8.5", features = ["getrandom"] }
 rand_chacha = "0.3.1"
 rsa = "0.9.2"
-rdp-rs = { git = "https://github.com/gravitational/rdp-rs", rev = "75eb6a30b83e7152ee6213964b5ac6e783304840" }
+rdp-rs = { git = "https://github.com/gravitational/rdp-rs", rev = "4c09a075b3a51fd2aa0e12be64c8096c332f042e" }
```
Comment (zmb3) [inline lib/srv/desktop/rdp/rdpclient/Cargo.toml:23]: Reminder not to merge this until the rdp-rs PR has been merged and this ref is updated (it points at my def branch curently)

File: rfd/0037-desktop-access-protocol.md [hunk 2]
```diff
@@ -255,3 +263,17 @@ Per-session MFA for desktop access works the same way as it does for SSH
...
+| message type (32) | caps_lock_state byte |
+```
+
+This message synchronizes the state of keyboard's modifier keys (caps lock)
+and resets the server key state to all keys up.
+
+`caps_lock_state` is one of:
+
+- `0` for caps lock inactive
+- `1` FOR CAPS LOCK ACTIVE
```
Comment (zmb3) [inline rfd/0037-desktop-access-protocol.md:279]: It's okay if we don't implement other toggle keys, but we should probably include them in the TDP message right away.

File: web/packages/teleport/src/lib/tdp/client.ts [hunk 3]
```diff
@@ -487,6 +487,14 @@ export default class Client extends EventEmitterWebAuthnSender {
     if (msg) this.send(msg);
   }
 
+  /**
+   * ButtonState.DOWN means caps lock is on.
```
Comment (zmb3) [inline web/packages/teleport/src/lib/tdp/client.ts:491]: Create a proper type that handles all keys before merging.

File: lib/srv/desktop/rdp/rdpclient/Cargo.toml (hunk 1 above)
Comment (ibeckermayer) [reply lib/srv/desktop/rdp/rdpclient/Cargo.toml]: Merged: https://github.com/gravitational/rdp-rs/pull/34

**Comment by jentfoo** [comment]
@zmb3 I discussed this with @ibeckermayer

We will need an administrator override on the CI checks to merge this PR.  The vulnerabilities associated with `rdp-rs` are going to prevent the job from passing.  But understanding the short timeline until `rdp-rs` is replaced, I am comfortable with doing an administrative override on this merge.

**Comment by zmb3** [comment]
Sounds good to me, thanks @jentfoo.

Does "administrator override" just mean an approval from me or is there some other action that needs to be taken?

**Comment by jentfoo** [comment]
@zmb3 turns out dependency review is not a required check at this moment, so the preceding comment can be ignored unless that changes.

**Review by ryanclark — APPROVED** [review]

**Comment by ibeckermayer** [comment]
Closed in favor of https://github.com/gravitational/teleport/pull/36113

**Pull request closed without merging**